agent/consul/rate/handler.go

```diff
@@ -175,14 +182,42 @@ func (h *Handler) Run(ctx context.Context) {
 // Allow returns an error if the given operation is not allowed to proceed
 // because of an exhausted rate-limit.
 func (h *Handler) Allow(op Operation) error {
+
+	if h.delegate == nil {
+		panic("delegate not set on handler via RegisterDelegate(..)")
```

Comment:
Maybe ass some logs around here to point that this should never happen and if it happen, it should not panic the whole program but should recover because we have a panic recovery handler added to the server.

Reply:
Added a TODO to follow through on the panic handler stuff (I'm not too familiar with recovery handlers atm) so as not to block this PR.  Updates in https://github.com/hashicorp/consul/pull/15857/commits/d86244e8f6f03a5ea5decd111a978bc59c31b4dc